File: Common/Engine/src/common/engine/components/menu/Menu.java
package common.engine.components.menu;

import javafx.scene.Scene;
import javafx.scene.layout.Pane;

import java.util.Observable;
import java.util.Observer;

public abstract class Menu extends Observable implements Observer {

    private Scene scene;

    public Menu() {
    }

    @Override
    public void notifyObservers() {
        this.setChanged();
        super.notifyObservers();
    }

    @Override
    public void notifyObservers(Object arg) {
        this.setChanged();
        super.notifyObservers(arg);
    }

    public Scene getScene() {
        return scene;
    }

    public void setMenuPane(Pane menuPane) {
        if (menuPane == null) {
            throw new NullPointerException("menuPane is null");
        }
        if (this.scene != null) {
            this.scene.setRoot(menuPane);
        } else {
            this.scene = new Scene(menuPane);
        }
    }

}

File: Common/Engine/src/common/engine/components/menu/Menu.kt
package common.engine.components.menu

import javafx.scene.Scene
import javafx.scene.layout.Pane
import java.util.*

abstract class Menu : Observable(), Observer {
    var scene: Scene? = null
        private set

    override fun update(o: Observable?, arg: Any?) {

    }

    override fun notifyObservers() {
        setChanged()
        super.notifyObservers()
    }

    override fun notifyObservers(arg: Any) {
        setChanged()
        super.notifyObservers(arg)
    }

    fun setMenuPane(menuPane: Pane?) {
        if (menuPane == null) {
            throw NullPointerException("menuPane is null")
        }
        if (scene != null) {
            scene!!.root = menuPane
        } else {
            scene = Scene(menuPane)
        }
    }
}
